inithome/src/rubin/nublado/inithome/provisioner.py

```diff
@@ -0,0 +1,103 @@
+"""Provisioner for user home directories."""
+
+import asyncio
+import logging
+from os import chown, environ, getenv, umask
+from pathlib import Path
+
+MAX_ID = 2**32 - 1  # True for Linux, which is where we run inithome
+
+
+class Provisioner:
+    """Object to oversee idempotent provisioning of home directories."""
+
+    def __init__(self, uid: int, gid: int, homedir: Path) -> None:
+        self._validate(uid, gid)
+        self.uid = uid
+        self.gid = gid
+        self.homedir = homedir
+
+    def _validate(self, uid: int, gid: int) -> None:
+        for item in (uid, gid):
+            if item == 0:
+                raise ValueError("Will not provision for UID/GID 0")
+            if item < 0:
+                raise ValueError("UID/GID must be positive")
+            if item > MAX_ID:
+                raise ValueError(f"UID/GID must be <= {MAX_ID}")
+
+    async def provision(self) -> None:
```

Comment:
I think you can just make this sync and save the `asyncio.run`. I can't imagine any case where we'd use async functions in this code.

Reply:
Oh, I can't either.  I just did that yesterday afternoon because I was peeved at Merlin.